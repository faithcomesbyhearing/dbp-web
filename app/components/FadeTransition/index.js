/**
 *
 * FadeTransition
 *
 */

import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';

function FadeTransition({ children, classNames, ...props }) {
	const nodeRef = useRef(null);
	return (
		<CSSTransition
			{...props}
			timeout={400}
			classNames={classNames || 'fade'}
			nodeRef={nodeRef}
		>
			{children}
		</CSSTransition>
	);
}

FadeTransition.propTypes = {
	children: PropTypes.node,
	classNames: PropTypes.string,
};

export default FadeTransition;
