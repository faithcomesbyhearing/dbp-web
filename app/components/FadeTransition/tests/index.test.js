import React from 'react';
import { render } from '@testing-library/react';
import FadeTransition from '..';

const children = (
	<div id="transition-child-id">
		<h1>Child of the transition</h1>
		<p>Paragraph child of the transition</p>
	</div>
);

describe('<FadeTransition /> component', () => {
	it('should match snapshot with classNames', () => {
		const { asFragment } = render(
			<FadeTransition classNames={'slide-from-right'}>
				{children}
			</FadeTransition>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
	it('should match snapshot without classNames', () => {
		const { asFragment } = render(<FadeTransition>{children}</FadeTransition>);
		expect(asFragment()).toMatchSnapshot();
	});
});
