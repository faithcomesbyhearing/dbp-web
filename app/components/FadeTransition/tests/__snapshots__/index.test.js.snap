// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FadeTransition /> component should match snapshot with classNames 1`] = `
<DocumentFragment>
  <div
    id="transition-child-id"
  >
    <h1>
      Child of the transition
    </h1>
    <p>
      Paragraph child of the transition
    </p>
  </div>
</DocumentFragment>
`;

exports[`<FadeTransition /> component should match snapshot without classNames 1`] = `
<DocumentFragment>
  <div
    id="transition-child-id"
  >
    <h1>
      Child of the transition
    </h1>
    <p>
      Paragraph child of the transition
    </p>
  </div>
</DocumentFragment>
`;
